File: src/main/java/itmo/is/project/model/module/production/ProductionModuleBlueprint.java
package itmo.is.project.model.module.production;

import itmo.is.project.model.module.ModuleBlueprint;
import jakarta.persistence.*;
import lombok.*;

import java.util.List;

@Entity
@Table(name = "production_module_blueprint")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class ProductionModuleBlueprint extends ModuleBlueprint {

    @OneToOne(mappedBy = "blueprint", cascade = CascadeType.ALL)
    private Production production;

    @OneToMany(mappedBy = "blueprint", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    private List<Consumption> consumption;
}

File: src/main/kotlin/itmo/isproject/model/module/production/ProductionModuleBlueprint.kt
package itmo.isproject.model.module.production

import itmo.isproject.model.module.BuildCost
import itmo.isproject.model.module.ModuleBlueprint
import jakarta.persistence.*
import java.util.*

@Entity
@Table(name = "production_module_blueprint")
class ProductionModuleBlueprint(

    @OneToOne(mappedBy = "blueprint", cascade = [CascadeType.ALL])
    var production: Production? = null,

    @OneToMany(mappedBy = "blueprint", fetch = FetchType.LAZY, cascade = [CascadeType.ALL])
    var consumption: MutableList<Consumption> = ArrayList(),

    buildCost: BuildCost? = null,
    name: String? = null,

) : ModuleBlueprint(buildCost, name)
